src/main/java/com/codeborne/selenide/commands/SetSelected.java

```diff
@@ -13,9 +13,23 @@
   public WebElement execute(SelenideElement proxy, WebElementSource locator, Object[] args) {
     boolean selected = (Boolean) args[0];
     WebElement element = locator.getWebElement();
-    if (element.isSelected() ^ selected) {
-      if (element.getAttribute("readonly") != null)
-        throw new InvalidStateException("Cannot change value of readonly element");
+    if (!element.isDisplayed()) {
+      throw new InvalidStateException("Cannot change invisible element");
+    }
```

Comment:
This is a valid change, but there is 1 thing I don't like. This code will call too many `WebElement` method, which makes it slow. `isDisplayed`, `getTagName`, `getAttribute("type")`, `getAttribute("readonly")`, `element.getAttribute("disabled")`. 

Probably we should implement it with one JS call?

Reply:
Btw why `element.isDisplayed()` check needs here? Doesn't Selenide check it automaticaly whatever?

Reply:
1. options->option fixed, Test added.
2. many getAttribute is actually not slow, because it just doesn't search at every call just once.
3. element.isDisplayed is needed, selenide did not check it at *this* place (not_existing_element.setSelected(false); -> works without error, which is wrong).